src/tests/test_spread.rs

```diff
@@ -139,3 +96,54 @@ fn spread_type_record() -> TestResult {
         "type_mismatch",
     )
 }
+
+#[test]
+fn spread_external_args() -> TestResult {
+    run_test(r#"^echo ...[1 "foo"] 2 ...[3 "bar"]"#, "1 foo 2 3 bar").unwrap();
```

Comment:
`echo` is traditionally not good to use in tests because cross-platform `echo`'s may work differently than expected. this is why we have things like `nu --testbin cococo 'test'`